src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,272 @@
+    if (previewContainer) {
+      this.updatePreviewContainer(previewContainer);
+    }
   }
 
-  private get shouldShowPreviewResults() {
-    return this.options.numberOfPreviewResults > 0;
+  private buildPreviewHeader(suggestion: string) {
+    const text = $$('span', {}, `${this.options.headerText} "${suggestion}"`).el;
+    const header = $$('div', { className: 'coveo-preview-header' }, text).el;
+    this.previewContainer.appendChild(header);
+  }
+
+  private buildResultsContainer() {
+    return $$('div', { className: 'coveo-preview-results' }).el;
+  }
+
+  private get previewContainer() {
+    return $$(this.root).find('.coveo-preview-container');
   }
 
   private querySuggestGetFocus(args: IQuerySuggestSelection) {
-    if (!this.shouldShowPreviewResults) {
+    if (!this.previewContainer) {
+      return;
+    }
+
+    if (this.previousSuggestionHovered === args.suggestion) {
       return;
     }
-    this.executeQueryHover(args.suggestion);
+
+    this.previousSuggestionHovered = args.suggestion;
+    this.timer && clearTimeout(this.timer);
+    this.timer = setTimeout(() => {
+      this.executeQueryHover(args.suggestion);
+    }, this.options.executeQueryDelay);
   }
 
-  private executeQueryHover(suggestion: string) {
+  private async executeQueryHover(suggestion: string) {
     const previousQueryOptions = this.queryController.getLastQuery();
     previousQueryOptions.q = suggestion;
     previousQueryOptions.numberOfResults = this.options.numberOfPreviewResults;
-    //TODO: I will need to execute a query, with the result,
-    //      build a container and display the result  next to the querySuggest
+    var results = await this.queryController.getEndpoint().search(previousQueryOptions);
+    $$(this.previewContainer).empty();
+    this.currentlyDisplayedResults = [];
+    if (!results) {
+      return;
+    }
+    this.addImage(results);
+    this.buildPreviewHeader(suggestion);
+    this.buildResultsPreview(results);
+  }
+
+  private async buildResultsPreview(results: IQueryResults) {
+    const resultsContainer = this.buildResultsContainer();
+    this.previewContainer.appendChild(resultsContainer);
+    this.setupRenderer(resultsContainer);
+    const HTMLResult = await this.templateToHtml.buildResults(results, 'preview', this.currentlyDisplayedResults);
+    if (!(HTMLResult.length > 0)) {
+      return;
+    }
+    this.updateResultPerRow(HTMLResult);
+    this.renderer.renderResults(HTMLResult, true, result => {});
+  }
+
+  private updateResultPerRow(elements: HTMLElement[]) {
+    let resultAvailibleSpace = '33%';
+    if (elements.length === 4) {
+      resultAvailibleSpace = '50%';
+    }
+    elements.forEach(element => {
+      this.updateFlexCSS(element, resultAvailibleSpace);
+    });
+  }
+
+  private updateFlexCSS(element: HTMLElement, value: string) {
+    element.style.flex = `0 0 ${value}`;
+  }
+
+  //Delete when creating the PR
+  private addImage(results: IQueryResults) {
```

Comment:
This whole method should not be in the final product... 

I understand it's there to test for the time being, but how can we make sure it does not slip past us ?